### DISCUSSION TITLE
Workflows with pull_request trigger never get triggered for PRs from forks

### DISCUSSION THREAD

**Discussion by danielkatz** [community/community#26981] [Actions]
<p>Hi,</p>
<p>When PRs get opened on our private repo from forks, then workflows with ‘pull_request’ trigger are not triggered.</p>
<p>This behavior is reproducible only on private repos.</p>
<p>We did some digging around:</p>
<p>1. Created a clean private repo <a href="https://github.com/elementor/tests-github-actions/pulls" rel="nofollow noopener">https://github.com/elementor/tests-github-actions/pulls</a> (now it’s public).</p>
<ol start="2">
<li>
<p>Added a <a href="https://github.com/elementor/tests-github-actions/blob/master/.github/workflows/blank.yml" rel="nofollow noopener">blank workflow</a> with <code>pull_request</code> as a trigger.</p>
</li>
<li>
<p>Edited README.md on a branch and created a <a href="https://github.com/elementor/tests-github-actions/pull/1" rel="nofollow noopener">PR</a> to master. :heavy_check_mark: The workflow ran.</p>
</li>
<li>
<p>Forked the repo into my account <a href="https://github.com/danielkatz/tests-github-actions" rel="nofollow noopener">https://github.com/danielkatz/tests-github-actions</a> (private).</p>
</li>
<li>
<p>On my fork, edited README.md on a master and created a <a href="https://github.com/elementor/tests-github-actions/pull/2" rel="nofollow noopener">second PR</a> to the original repo. :x: The workflow didn’t run.</p>
</li>
<li>
<p>We made the original repo public.</p>
</li>
<li>
<p>Again on the original repo, we edited README.md on a branch and created a <a href="https://github.com/elementor/tests-github-actions/pull/3" rel="nofollow noopener">third PR</a> to master. :heavy_check_mark: The workflow ran.</p>
</li>
<li>
<p>Again, forked the repo into my account <a href="https://github.com/danielkatz/tests-github-actions-1" rel="nofollow noopener">https://github.com/danielkatz/tests-github-actions-1</a> (public).</p>
</li>
</ol>
<p>9. On my fork, edited README.md on a master and created a <a href="https://github.com/elementor/tests-github-actions/pull/4" rel="nofollow noopener">forth PR</a> to the original repo. ✔ The workflow ran.</p>
<p>I am aware of the issues around a <a href="https://github.community/t/run-a-github-action-on-pull-request-for-pr-opened-from-a-forked-repo/16054">malicious PR mitigation</a>, but if this is the root cause of the observed behavior, why the workflow did trigger on a public repo? And more, the PRs to the private repo were made by users with permissions for that repo, thus should be considered as trusted.</p>
<p>Really appreciate help on this issue.</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi,</p>
<p>Many thanks for your feedback!  </p>
<p>This is by designed. If your base repository is private, workflow will not run when you open a pull request from forked repository. It’s mentioned <a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/events-that-trigger-workflows#pull-request-events-for-forked-repositories" rel="nofollow noopener">here</a>, please kindly check.</p>
<p>Here is same <a href="https://github.community/t/github-workflow-not-running-from-pull-request-from-forked-repository/16379">ticket</a> about this, please check more details. Hope it helps.</p>

**Comment by danielkatz** [comment]
<p>Thanks, I suspected this much…</p>
<p>Did you see the end of my post, I’m proposing to remove this restriction in case of a PR from a user that has privileges on the base repository. Since he should be considered a trusted user, and besides, if he would like to do something malicious, he could do it directly on the base repository.</p>
<p>Where can I file this suggestion as a feature request?</p>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/danielkatz">@danielkatz</a> ,</p>
<p>This has been submitted as a feature request already :slight_smile:</p>
<p>From github :</p>
<p>We have work going on to enable that scneario but the changes where deeper than just the actions token for a number of reasons.  I expect we will ship that before the end of the year.</p>

**Comment by chiefjester** [comment]
<p>I know it was submitted and I’m one those people trying to wait/push this. My 2 cents, it’s currently working on a public repo, the more reason it should work on a private repo. In a private repo setting, you are explicitly giving the user permission to your repository so chances of bad actors are smaller than in public repo settings. I don’t want to downplay the amount of work or deeper changes this one require but why would it take a much significant time if it’s already enabled on a public repo? The only difference between a public repo and a private repo is the visibility, what am I missing? :thinking:</p>

**Comment by FabianKoestring** [comment]
<p>Sorry the stupid question but what year do you mean? It feels like github doesn’t consider this problem important. I think a lot of paying customers are waiting for this feature.</p>
